crypto/err/err.c

```diff
@@ -23,9 +23,25 @@
 #include "internal/constant_time_locl.h"
 #include "e_os.h"
 
-static int err_load_strings(const ERR_STRING_DATA *str);
+#define ERR_FLAG_MARK      0x01
+#define ERR_FLAG_CLEAR     0x02
+#define ERR_NUM_ERRORS     16
+
+typedef struct err_entry_st {
+    int flags;
+    unsigned long buffer;
+    char *data;
+    int data_flags;
+    const char *file;
+    int line;
+} ERR_ENTRY;
+
+struct err_state_st {
+    ERR_ENTRY entries[ERR_NUM_ERRORS];
+    int top, bottom;
+};
```

Comment:
This part, I have to say I *really* like.  That thing with several parallel arrays was...  jarring